Frag grenade buffs

Kell-E opened the pull request:
Currently frag grenades suffer from multiple issues, they very unreliably are capable of hitting anyone and even when they do hit their low damage means they are almost never going to be able to pierce even thick clothing.

With these changes all types of fragmentation grenades now fill their intended purpose, being within a tile of an exploding grenade is likely to put you instantly into critical condition but with armor and distance you'll likely only take a few grazing hits.

The shrapnel was given some penetration so that it can challenge basic armor without shredding unarmored targets more than it already does. 
The values remain low enough that a medium plate will catch basically all of the incoming damage while the ballistic armor will make you more or less immune to (most) of a grenades effects (the explosion is still likely to knock you out for a few seconds)

Lastly, I gave shrapnel the ability to embed into targets so that targets in specialized armor still have to worry about embedded shrapnel into their exposed hands / feet. 

It's worth noting that if this is accepted, I'll PR an increase to the cost of fragmentation grenades and likely remove the free box that spawns in the merc base.

:cl: Kell-E
balance: All kinds of fragmentation grenades are now significantly more lethal
/:cl:

SierraKomodo reviewed (commented):
The high number of projectiles spawned by frag grenades cause lag spikes in the server. Standard frag grenades aren't as noticeable, currently, but the higher yield ones, pre-change, are already very noticeable when they go off.

Uncertain I want to approve increased projectile counts until that specific lag issue is addressed and resolved.

Kell-E commented:
Not something I'd considered, I'll see if I can figure out some way to drop the count but retain the same level of effectiveness

Kell-E committed:
Frag grenade buffs

Currently frag grenades suffer from multiple issues, they very unreliably are capable of hitting anyone and even when they do hit their low damage means they are almost never going to be able to pierce even thick clothing.

With these changes all types of fragmentation grenades now fill their intended purpose, being within a tile of an exploding grenade is likely to put you instantly into critical condition but with armor and distance you'll likely only take a few grazing hits.

The shrapnel was given some penetration so that it can challenge basic armor without shredding unarmored targets more than it already does. 
The values remain low enough that a medium plate will catch basically all of the incoming damage while the ballistic armor will make you more or less immune to (most) of a grenades effects (the explosion is still likely to knock you out for a few seconds) 
Increases grenade efficiency, decreases lag
